### PR TITLE
Reuse GenerationConfig

### PR DESCRIPTION
**Pull request by Wovchena**

### CODE REVIEW

File: tests/python_tests/continuous_batching/common.py
Comment (Wovchena) [inline tests/python_tests/continuous_batching/common.py:1]: @popovaan, @iefode, @mzegla, python tests fail wiht
```
>       output = pipe.generate(prompts, generation_configs)
E       RuntimeError: Exception from src\inference\src\cpp\infer_request.cpp:223:
E       Exception from src\plugins\intel_cpu\src\graph.cpp:1367:
E       Node PagedAttentionExtension_86689 of type PagedAttentionExtension
E       Check 'S % 16 == 0' failed at build\src\plugins\intel_cpu\cross-compiled\AVX2\executor_pa.cpp:1469:
E       CPU: head size must be multiple of 16, current: 8
```
but `/c/Users/vzlobin/r/openvino/bin/intel64/Release/throughput_benchmark.exe -m TinyLlama-1.1B-Chat-v1.0/ --dataset 'c:/Users/vzlobin/Downloads/ShareGPT_V3_unfiltered_cleaned_split.json' --request_rate 1 --dynamic_split_fuse` passes. Do I need to rebuild openvino with any specific flag?

File: samples/cpp/throughput_benchmark/throughput_benchmark.cpp [hunk 1]
```diff
@@ -82,6 +82,18 @@ struct Dataset {
     }
 };
 
+ov::genai::GenerationConfig greedy() {
```
Comment (iefode) [inline samples/cpp/throughput_benchmark/throughput_benchmark.cpp:85]: Minor: we can move such config examples to separated header shared between all samples not to duplicate code

File: src/cpp/include/openvino/genai/generation_config.hpp [hunk 2]
```diff
@@ -66,6 +70,7 @@ class OPENVINO_GENAI_EXPORTS GenerationConfig {
     size_t max_new_tokens = SIZE_MAX;
     size_t max_length = SIZE_MAX;
     bool ignore_eos = false;
+    size_t min_new_tokens = 0;
```
Comment (iefode) [inline src/cpp/include/openvino/genai/generation_config.hpp:73]: Looks like support of `num_return_sequence` is lost. Please do not forget about this parameter

File: tests/python_tests/continuous_batching/common.py
Comment (iefode) [reply tests/python_tests/continuous_batching/common.py:1]: @Wovchena Which tests do you mean?

Comment (Wovchena) [reply tests/python_tests/continuous_batching/common.py:1]: `test_sampling_precommit[model_id=hf-tiny-model-private/tiny-random-CodeGenForCausalLM]` is the first failing that way when I run `cmake -DENABLE_CONTINUOUS_BATCHING=y -DCPACK_ARCHIVE_COMPONENT_INSTALL=n -DOpenVINO_DIR=/c/Users/vzlobin/r/openvino/build/ -B build . && CMAKE_BUILD_PARALLEL_LEVEL= cmake --build build --config Release && PATH=/c/Users/vzlobin/r/openvino/temp/tbb/bin/:/c/Users/vzlobin/r/openvino/bin/intel64/Release/Release/:/c/Users/vzlobin/r/g/build/openvino_genai/:$PATH OPENVINO_LIB_PATHS=/c/Users/vzlobin/r/openvino/temp/tbb/bin/:/c/Users/vzlobin/r/openvino/bin/intel64/Release/ PYTHONPATH=/c/Users/vzlobin/r/openvino/bin/intel64/Release/python/:/c/Users/vzlobin/r/g/build/ python -m pytest tests/python_tests/continuous_batching/test_sampling.py -s --sw`

File: src/cpp/src/generation_config.cpp [hunk 3]
```diff
@@ -96,7 +96,7 @@ void GenerationConfig::validate() const {
                     "Beam search with sampling is not supported yet. "
                     "Please either set do_sample=false to use beam search "
                     "or set num_beams=1 if you with to use multinomial sampling.");
-    OPENVINO_ASSERT(num_return_sequences <= num_beams, "num_return_sequences must be less or equal to num_beams");
+    OPENVINO_ASSERT(num_return_sequences > 0, "num_return_sequences must be greater than 0");
```
Comment (iefode) [inline src/cpp/src/generation_config.cpp:99]: Please, insert the lost parameters from here: https://github.com/ilya-lavrenov/openvino.genai/blob/e7562739ecfd1407d67cedd28d8933f791498d51/text_generation/causal_lm/cpp/continuous_batching/library/src/generation_config.cpp#L22

File: src/cpp/continuous_batching/src/continuous_batching_pipeline.cpp [hunk 4]
```diff
@@ -115,8 +115,8 @@ class ContinuousBatchingPipeline::Impl {
         return m_tokenizer;
     }
 
-    GenerationHandle add_request(uint64_t request_id, std::string prompt, GenerationConfig sampling_params) {
-        sampling_params.set_eos_token_id(m_tokenizer->get_eos_token_id());
+    GenerationHandle add_request(uint64_t request_id, std::string prompt, ov::genai::GenerationConfig sampling_params) {
+        sampling_params.eos_token_id = m_tokenizer->get_eos_token_id();
```
Comment (iefode) [inline src/cpp/continuous_batching/src/continuous_batching_pipeline.cpp:119]:
```
        OPENVINO_ASSERT(eos_token_id == tokenizer_eos_token_id,
            "EOS token ID is different in generation config (", eos_token_id, ") and tokenizer (",
            tokenizer_eos_token_id, ")");
```

**Review by iefode — APPROVED** [review]

**Review by mzegla — APPROVED** [review]

**Review by popovaan — APPROVED** [review]

**Review by pavel-esir — APPROVED** [review]

File: src/cpp/continuous_batching/src/sampler.hpp [hunk 5]
```diff
@@ -110,14 +110,17 @@ struct Group {
...
+    int64_t finish(Beam beam, const ov::genai::GenerationConfig& sampling_params) {
         int64_t preeempted_sequence_id = -1;
         float generated_len = beam.get_generated_len() + (beam.m_token_id == sampling_params.eos_token_id ? 1 : 0); // HF counts EOS token in generation length
         beam.m_score /= std::pow(generated_len, sampling_params.length_penalty);
 
         min_heap.push_back(beam);
         std::push_heap(min_heap.begin(), min_heap.end(), greater);
-        if (min_heap.size() > sampling_params.group_size) {
+        OPENVINO_ASSERT(sampling_params.num_beams % sampling_params.num_beam_groups == 0,
+            "number of beams should be divisible by number of groups");
```
Comment (ilya-lavrenov) [inline src/cpp/continuous_batching/src/sampler.hpp:121]: such checks must be in validation of generation config. If they present there, let's drop from current places

File: src/cpp/continuous_batching/src/sampler.hpp [hunk 6]
```diff
@@ -126,24 +129,27 @@ struct Group {
         return preeempted_sequence_id;
     }
 
-    void is_done(const GenerationConfig& sampling_params) {
-        if (min_heap.size() < sampling_params.group_size)
+    void is_done(const ov::genai::GenerationConfig& sampling_params) {
+        OPENVINO_ASSERT(sampling_params.num_beams % sampling_params.num_beam_groups == 0,
+            "number of beams should be divisible by number of groups");
```
Comment (ilya-lavrenov) [inline src/cpp/continuous_batching/src/sampler.hpp:134]: the same

File: src/cpp/continuous_batching/src/sampler.hpp [hunk 7]
```diff
@@ -360,13 +366,16 @@ SamplerOutput Sampler::sample(std::vector<SequenceGroup::Ptr> & sequence_groups,
 GroupBeamSearcher::GroupBeamSearcher(SequenceGroup::Ptr sequence_group)
     : m_sequence_group(sequence_group),
         m_parameters{m_sequence_group->get_sampling_parameters()},
-        m_groups{m_parameters.num_groups} {
+        m_groups{m_parameters.num_beam_groups} {
     OPENVINO_ASSERT(m_sequence_group->num_running_seqs() == 1);
+    OPENVINO_ASSERT(m_parameters.num_beams % m_parameters.num_beam_groups == 0,
+        "number of beams should be divisible by number of groups");
```
Comment (ilya-lavrenov) [inline src/cpp/continuous_batching/src/sampler.hpp:372]: again

File: src/cpp/continuous_batching/src/sampler.hpp [hunk 8]
```diff
@@ -375,10 +384,13 @@ GroupBeamSearcher::GroupBeamSearcher(SequenceGroup::Ptr sequence_group)
 }
 
 void GroupBeamSearcher::select_next_tokens(const ov::Tensor& logits, SamplerOutput& sampler_output) {
+    OPENVINO_ASSERT(m_parameters.num_beams % m_parameters.num_beam_groups == 0,
+        "number of beams should be divisible by number of groups");
```
Comment (ilya-lavrenov) [inline src/cpp/continuous_batching/src/sampler.hpp:388]: as I see - it's not part of generation config validation, let's move it there and remove from sampler code.

**Pull request merged**